### DISCUSSION TITLE
Compatibility with GitLab CI

### DISCUSSION THREAD

**Discussion by Belco90** [ossjs/release#43] [Ideas]
Hi there! I'd like to ask about the possibility of making this awesome package compatible with GitLab CI.

**Comment by kettanaito** [comment]
Hey, @Belco90. 

Not sure that's an easy endeavor, given how much we currently rely on GitHub API (checking issues/pull requests/creating releases/etc). GitLab CI support, basically, means rewriting this library from scratch. 

If you wish to team up with me on this, I'd be totally for it, but I won't be doing this by myself.

**Reply by Belco90** [reply]
Thanks for your quick answer! I'm not that expert with GitLab API yet, but I'll probably have to work with it soon at my job. When I feel confident enough about it, I'll get back to this discussion to team up on this!

About rewriting the library from scratch: do you think it could be possible to extract some sort of core, then publish a specific package for GitLab? I haven't checked the source code yet, so I may be saying something completely stupid.

Last but not least: if you feel like won't like to maintain GitLab CI compatibility and keep it simple with just GitHub-only integration, please say so!

**Reply by kettanaito** [reply]
My intention with this package was to create a very opinionated experience focused on GitHub and how I think packages should be published. I do want more people to have sane publishing automation but at the same time I'm unlikely to maintain a Swiss knife kind of library just because of that. 

We can certainly abstract some pieces of logic from the current package, turn it into a monorepo or something, and then maybe have plugins but even at this stage it feels like adding so much complexity. I don't know if I'm sold on this, to be completely frank. It also depends vastly on how GitLab works and what API they have. Maybe it'd be a matter of swapping get/create release endpoints or maybe it'd mean an entire rewrite just to suit GitLab. 

You can certainly look into it, read about it and experiment. If you do, please share your findings so we know how to move forward.

**Reply by Belco90** [reply]
Hi again! I found another package that works for GitLab CI, so this idea isn't needed anymore. Feel free to close, and thanks for your help!
